core/src/main/java/hudson/slaves/Cloud.java

```diff
@@ -322,11 +389,13 @@ public HttpResponse doConfigSubmit(StaplerRequest req, StaplerResponse rsp) thro
         if (cloud == null) {
             throw new ServletException("No such cloud " + this.name);
         }
-        Cloud result = cloud.reconfigure(req, req.getSubmittedForm());
-        String proposedName = result.name;
-        if (!proposedName.equals(this.name)
-                && j.getCloud(proposedName) != null) {
-            throw new Descriptor.FormException(jenkins.agents.Messages.CloudSet_CloudAlreadyExists(proposedName), "name");
+        JSONObject formData = req.getSubmittedForm();
+        // add the cloud name to the submitted form data
+        formData.put("name", name);
+        Cloud result = cloud.reconfigure(req, formData);
+        if (!(result.name).equals(this.name)) {
+            // Do not rename the cloud in the config page. Use doConfirmRename()
```

Comment:
Who is the intended audience for this imperative statement? If this is temporary logic to support some sort of migration period, then a warning should be logged that the cloud is using an unsupported mechanism that will be eventually removed in the future. (And if so, that also raises the question, why support this but not creating a new cloud with a plugin that still uses the configure page during the same migration period?)

Reply:
Well, originally I had it as a form exception and I just wanted to block old/incorrect implementations in the downstream plugins' `config.jelly`. My rationale for a quiet ignore was that there is a large sidebar link that says rename.